### PR TITLE
refactor(tree, tree-item): allow all arrow keydown events to propagate outside the root tree element

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #4158 

## Summary

This PR refactors `calcite-tree` and `calcite-tree-item` to not only fix the keyboard behavior for focusing within trees and subtrees, but it also allows all of the native "Arrow*" `keydown` events to propagate outside the containing root tree, making it more predictable and consistent with native behavior.

### CODE REVIEW

File: src/components/tree/tree.e2e.ts [hunk 1]
```diff
@@ -349,4 +349,284 @@ describe("calcite-tree", () => {
       });
     });
   });
+
+  describe("keyboard support", () => {
```
Comment (jcfranco) [inline src/components/tree/tree.e2e.ts:353]: Thanks for adding these tests! ⌨️ 👀

File: src/components/tree/tree.tsx [hunk 2]
```diff
@@ -230,6 +230,59 @@ export class Tree {
...
   }
 
+  @Listen("keydown")
+  keyDownHandler(event: KeyboardEvent): void {
+    const root = this.el.closest("calcite-tree:not([child])") as HTMLCalciteTreeElement;
+    const target = event.target as HTMLCalciteTreeItemElement;
+
+    if (root === this.el && target.tagName === "CALCITE-TREE-ITEM" && this.el.contains(target)) {
+      switch (event.key) {
+        case "ArrowDown":
```
Comment (jcfranco) [inline src/components/tree/tree.tsx:240]: Can you call `preventDefault` wherever the component handles the event?

**Review by jcfranco — COMMENTED** [review]
Looking good! Would it be possible to move all keyboard event handling to the parent tree and not have the navigation/selection code in two different places?

**Comment by eriklharper** [comment]
> Looking good! Would it be possible to move all keyboard event handling to the parent tree and not have the navigation/selection code in two different places?

Yeah, I can create a follow-up refactor issue for that, unless you think its within the scope of this ticket?

**Comment by jcfranco** [comment]
I think we can keep it within this ticket since the code is related (event propagation and navigation code moving to tree), right?

**Review by jcfranco — APPROVED** [review]
Sweet! 🍬

**Comment by jcfranco** [comment]
@benelan @anveshmekala @Elijbet could you help do some testing on this to make sure tree is behaving as expected? We're slowly ramping up automated tests for tree/tree-item, so manual testing would be helpful until we have better coverage.

**Comment by anveshmekala** [comment]
> I think we can keep it within this ticket since the code is related (event propagation and navigation code moving to tree), right?

Testing looks good 🥳

**Pull request merged**